graalvm-native-image-app/src/main/com/mongodb/internal/graalvm/NativeImageApp.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.internal.graalvm;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+import static com.mongodb.ClusterFixture.getConnectionStringSystemPropertyOrDefault;
+
+final class NativeImageApp {
+    private static final Logger LOGGER = LoggerFactory.getLogger(NativeImageApp.class);
+
+    public static void main(final String[] args) {
+        LOGGER.info("java.vendor={}, java.vm.name={}, java.version={}",
+                System.getProperty("java.vendor"), System.getProperty("java.vm.name"), System.getProperty("java.version"));
+        String[] arguments = new String[] {getConnectionStringSystemPropertyOrDefault()};
```

Comment:
If I run like

```
./graalvm-native-image-app/build/native/nativeCompile/NativeImageApp -Dorg.mongodb.test.uri=mongodb://localhost:37017/?serverselectiontimeoutms=500
```

When no server is running on port 37017, it eventually hangs.  I'm not sure which of the examples is causing that.

But if I run with Gradle, like

```
env JAVA_HOME="${JDK17}" ./gradlew -PjavaVersion=21 :graalvm-native-image-app:nativeRun -Dorg.mongodb.test.uri=mongodb://localhost:37017/?serverselectiontimeoutms=500
```

then the system properties appears to be ignored.  The build is marked as successful and I can see in the logs that it's connecting to localhost:27017.  That indicates to me that in Evergreen the system property is likely being ignored as well.

Reply:
Turns out, the Java system properties passed this way have to propagated manually. We do that in `./build.gradle` for `Test` Gradle tasks.

Fixed in https://github.com/mongodb/mongo-java-driver/pull/1338/commits/3c92447fa4f300435d051e9405247c986d4ba139.